src/ansiblelint/rules/NoTabsRule.py

```diff
@@ -29,10 +29,15 @@ class NoTabsRule(AnsibleLintRule):
     def matchtask(
         self, task: Dict[str, Any], file: 'Optional[Lintable]' = None
     ) -> Union[bool, str]:
-        for k, v, parent in nested_items(task):
+        for k, v, parent_path in nested_items_path(task):
             if isinstance(k, str) and '\t' in k:
                 return True
-            if (parent, k) not in self.allow_list and isinstance(v, str) and '\t' in v:
+            if (
+                parent_path
+                and (parent_path[-1], k) not in self.allow_list
```

Comment:
I'd suggest putting `parent_path[-1]` into a variable. It's currently unclear what this is on this level of abstraction

Reply:
done. Is it clearer now?

Reply:
Yep, looks better.